File: src/main/java/com/nanaios/polygonal_tech/registries/impl/DeferredItemRegister.java
package com.nanaios.polygonal_tech.registries.impl;

import com.nanaios.polygonal_tech.registries.base.WrapperDeferredRegister;
import com.nanaios.polygonal_tech.registries.interfaces.IDisplayable;
import net.minecraft.world.item.CreativeModeTab;
import net.minecraft.world.item.Item;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.RegistryObject;

public class DeferredItemRegister extends WrapperDeferredRegister<Item> implements IDisplayable {
    public DeferredItemRegister(DeferredRegister<Item> deferredRegister) {
        super(deferredRegister);
    }

    @Override
    public void display(CreativeModeTab.Output output) {
        for(RegistryObject<Item> item : getEntries()) {
            output.accept(item.get());
        }
    }
}

File: src/main/kotlin/com/nanaios/polygonal_tech/lib/register/DeferredItemRegister.kt
package com.nanaios.polygonal_tech.lib.register

import net.minecraft.world.item.Item
import net.minecraftforge.registries.DeferredRegister
import net.minecraftforge.registries.ForgeRegistries

open class DeferredItemRegister(modId: String): DeferredExtendRegister<Item>(DeferredRegister.create(ForgeRegistries.ITEMS,modId)) {
}